rts/lra/lra-service-base/src/main/java/io/narayana/lra/LRAHttpClient.java

```diff
@@ -0,0 +1,64 @@
+package io.narayana.lra;
+
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.FutureRequestExecutionService;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.client.HttpRequestFutureTask;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class LRAHttpClient {
+    public static final long PARTICIPANT_TIMEOUT = 1; // number of seconds to wait for requests
+
+    private static LRAHttpClient client = new LRAHttpClient();
+
+    private FutureRequestExecutionService futureRequestExecutionService;
+
+    private LRAHttpClient() {
+        HttpClient httpClient = HttpClientBuilder.create().setMaxConnPerRoute(5).build();
+        ExecutorService executorService = Executors.newFixedThreadPool(5);
```

Comment:
just an idea: maybe we can make these values (maxConnPerRoute and fixedThreadPool) configurable in the future?

Reply:
I agree with @xstefank - unless there is a good reason for them to be fixed I think they should be configurable. Is there an LRAEnvironmentBean, if not please can you raise a JBTM to add one and mention these values so we don't miss them

Reply:
Perhaps in the future but let's wait for when we add metrics and non functional testing

Reply:
So to track it I created JBTM-3213

Reply:
Thanks, that works for me